### DISCUSSION TITLE
What are typical Fediverse expectations and behaviors surrounding rule-breaking federated content?

### DISCUSSION THREAD

**Discussion by afontenot** [mastodon/mastodon#21914] [General]
This is something I've been puzzled by in my time on Mastodon and I wanted to open this up to a very general discussion about what others have observed and hopefully get a few opinions from long-time users. I suspect this is something even more confusing for new users who don't really understand federation, so it would be cool if this discussion led to clearer communication on the subject.

Let's take a simple case - content warnings. My instance has the rule "Sexually explicit or violent media must be marked as sensitive when posting." But it federates with servers that to the best of my knowledge do not have that rule, or else enforce it with very different norms for what counts as "sexually explicit" than are applied on my home instance.

How am *I*, as a user on an instance where this content requires warnings, supposed to (or allowed to) relate to this content? Should I (can I)

 * Follow users on these instances? This would result in introducing content into the federated timeline that would be rule breaking if it was posted on my instance.
 * Boost or comment on content that would be rule breaking if it were posted on my instance?

Likewise, let's say that I joined the server I did *because* it takes a strong approach towards content warnings on sexually explicit material. Suppose I encounter some federated content that breaks this rule. Should I report it? What action, if any, should I expect my instance's moderators to take? What action do they have *the capacity* to take for federated content? Do I need to read the rules for a federated instance before I check the box to anonymously forward a report to that instance's moderators?

Last, let's say I am someone on a server with loose rules around content warnings. When I'm interacting with some federated content from an instance that happens to have stronger rules, surely I'm not required to know the remote instance's expectations before interacting. So what exactly *are* the expectations on me? Is it possible to get in trouble with my own instance's moderators without breaking a single rule of my instance, just because I annoyed the wrong people on a federated server?

The same questions and considerations apply *mutatis mutandis* to other rules typical of Mastodon servers. E.g. something like ~1/3 of the popular Mastodon servers have no rule against spam; does that become a problem when spam becomes visible on timelines that disallow it? I assume if I join an instance that disallows spam and then follow 500 spambots on mastodon.social, dragging their content into the federated timeline, I'm going to get banned for that. But which rule did I break? It's (worryingly) hard to say in most cases.

----------------------------------------

I want to emphasize that these are just the questions that *I* have. I don't want anyone to feel like they have to have answers to all (or any) of them in order to reply. Questions of your own are also great, as are related opinions and observations.

The biggest concern I have is that I don't think I've seen a single set of instance rules that answered any of the above questions. This just doesn't seem to be on the radar for most admins. They strike me as pretty reasonable things to want to know about Mastodon and the Fediverse more generally, and I expect a lot of new users have questions about how moderation works in practice across federated instances.

**Comment by NSFWHubSocial** [comment]
I subscribed to this thread and I'm hoping to hear about this as well. My instance will be dedicated to NSFW content and I have been considering asking if there is a way to configure user's posts to automatically be local only (not Federated) unless like minded instances decide to connect with me. As a courtesy to the Fediverse, I would like for posts to NOT propagate to the Fediverse automatically. I would also like for instance admins to not have to take time to block me. I do indeed love the openness of the Fediverse philosophy and ideology and I would love to connect with everyone, but I also want to be courteous and respectful. Not everyone will be interested in the content on my instance so I will keep an eye on this thread to see if there is a good game plan on how best to go about setting up.

**Reply by nightpool** [reply]
What do you mean by "propagate to the fediverse automatically"? Unless a remote user follows an account on your instance, no information is sent to other servers. Information is only ever sent to users that follow your account (or, if you're replying to another user, the users that follow them)

**Reply by hansw-nlo** [reply]
> What do you mean by "propagate to the fediverse automatically"? Unless a remote user follows an account on your instance, no information is sent to other servers. Information is only ever sent to users that follow your account (or, if you're replying to another user, the users that follow them)

Post one public available post to Activity pub and a sh*tload of Mastodon instances will come and fetch it, not only the person you are sending it to.

**Comment by hansw-nlo** [comment]
What some can or can't do is not important since that often is a personal choice ( a good thing imho ). I can state that some on the fediverse have tools to filter content on all kinds of levels. Hubzilla allows me to filter away a lot I do not want to see while still being able to join in a discussion on topics I can see. 

Some examples. 

Forward/Reposts are in general filtered away since there are some 20 people in my connections that reshare everything they can lay their hands on, mostly (imho) stupid meme's. I simply do not see them, they do not get imported. 

#tags

Same idea as above. Some people in the USA talk about politics a lot. I do not see those posts if specific tags or words in the content are used. If there is only one topic they sometimes talk about and I do not care about the rest then I can see only posts on that topic (one might miss something in that case). 

Server level, 

I think most can filter on a server level. 


As for nsfw, we have content warnings. I can only advise to use the #nsfw tag in that case.
